SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -50,20 +61,46 @@ function Get-Secret
         $Name = $Matches[1]
     }
 
-    $res = Invoke-lpass 'show','--name', $Name, '--password'
-    if ([string]::IsNullOrWhiteSpace($res)) {
-        $res = Invoke-lpass 'show', '--name', $Name, '--notes'
-    } else {
-        # We have a password, check for a username
-        $username = Invoke-lpass 'show', '--name', $Name, '--username'
-        if ($username) {
-            $res = [System.Management.Automation.PSCredential]::new(
-                $username,
-                (ConvertTo-SecureString $res -AsPlainText -Force))
-        }
+    try {
+        $res = Invoke-lpass 'show', '--name', $Name, '--all'
+    }
+    catch {
+        Write-Error $_
+        return 
     }
+    
+    $Raw = ($res | Select-Object -Skip 1) -join "`n"
+
+    # Custom type cannot have the same case-sensitive name but Username / uSERname work.
+    # That's why we don't parse it directly to a hashtable.
+    $MyMatches = @([regex]::Matches($raw, '(?<key>.*?)\: (?<value>.*?)(?:\n|$)')  | ForEach-Object {
+            [PSCustomObject]@{
+                key        = $_.Groups.Item('key').value
+                value      = $_.Groups.Item('value').value 
+                valueIndex = $_.Groups.Item('value').index
+            }
+        })
+
+        if ([String]::IsNullOrEmpty($Raw)) {
+            Write-Error 'Unable to retrieve secret. Make sure you are logged in and try again.'
+            return ""
+        }
 
-    return $res
+    # Notes is always the last item. This is also the only field that can be multiline.
+    $HasNote = $MyMatches.key -ccontains 'Notes' 
+    if ($HasNote) {
+        $start = $MyMatches.Where({$_.Key -ceq 'Notes'},'Last')[0].valueIndex
+        $Note = $raw.Substring($start)
+    }
+    $IsCustomType = $AdditionalParameters.outputType -eq 'Detailed' -or $MyMatches.key.Contains('NoteType')
+    If ($IsCustomType) {
+        $Output = Get-ComplexSecret -Fields $MyMatches -Note $Note -Raw $Raw
+    }
+    else {
+        $Output = Get-SimpleSecret -Fields $MyMatches -Note $Note
```

Comment:
That's what make the distinction between 'Detailed' and the default. I preserved the default behavior of returning the Notes as string and regular user / password as psCredential.